llvm/include/llvm/CodeGen/TargetSubtargetInfo.h

```diff
@@ -323,6 +324,12 @@ class TargetSubtargetInfo : public MCSubtargetInfo {
   /// helps removing redundant copies generated by register allocator when
   /// handling complex eviction chains.
   virtual bool enableSpillageCopyElimination() const { return false; }
+
+  /// Enable macro fusion for this subtarget.
+  virtual bool enableMacroFusion() const { return false; }
```

Comment:
It's virtual here

Reply:
This function will be overrided in TableGen-generated XXXGenSubtargetInfo classes if subtarget defines some fusions. I think it should be virtual.

Reply:
Where is the use context where enableMacroFusion is necessary? I assume the use just directly iterates through the returned instances from getMacroFusions. That is, with something like:

```
if (enableMacroFusion()) 
  doStuff(getMacroFusions()) 

```

there's no point in having enableMacroFusion at all

Reply:
Oh, I get it. The typical usage is like:
```
    if (ST.enableMacroFusion()) {
      ScheduleDAGMI *DAG = createGenericSchedPostRA(C);
      DAG->addMutation(createMacroFusionDAGMutation(ST.getMacroFusions()));
      return DAG;
    }
```
`enableMacroFusion()` is for knowing if we should enable MacroFusion, `getMacroFusions()` returns the predicates.

Yes, I know we can replace `enableMacroFusion()` with `!getMacroFusions().empty()`, but the implementation of `enableMacroFusion()` avoid the creation of `std::vector`.

Let't take RISCV as an example: https://github.com/llvm/llvm-project/pull/72224/commits/1c92927e7d46c2b3c3f173e35afb2eecd4e9ffbf.
`enableMacroFusion()` is a replacement of `RISCVSubtarget::hasMacroFusion()`:
```
bool hasMacroFusion() const {
    return hasLUIADDIFusion() || hasAUIPCADDIFusion() || hasZExtHFusion() ||
           hasZExtWFusion() || hasShiftedZExtWFusion() || hasLDADDFusion();
}
```

Reply:
But you're just avoiding the creation of an empty vector, which doesn't matter. I would just do
vector Mutations = ST.getMacroFusions()
if (!Mutations.empty())
  DAG->addMutation(createMacroFusion(std::move(Mutations))

Reply:
OK, I will remove `enableMacroFusion()`.